scripts/ci/check-secrets-resolver.sh

```
@@ -0,0 +1,59 @@
+
+# ── SR-9: sample completeness (by_flavor expanded; env_var + consume_at exempt) ──
+ruby -ryaml -e '
+  y=YAML.load_file("'"$LAYOUT"'"); root=y["roots"]["sample"]; miss=[]
+  y["secrets"].each{|k,s| next if s["kind"]=="env_var" || s["consume_at"]
+    rels = s["by_flavor"] ? s["by_flavor"].values.map{|v|v["rel"]} : [s["rel"]]
+    rels.compact.each{|r| p=File.join(root,r); miss<<p unless File.exist?(p)} }
+  unless miss.empty?; warn "❌ SR-9: missing sample files:\n  #{miss.join("\n  ")}"; exit 1; end
+' || fail=1
+
+# ── SR-10: no real secrets in sample/ (placeholder-marked files are exempt) ──
+for f in $(grep -rlE 'sk_live_|sk_test_|-----BEGIN .*PRIVATE KEY-----' secrets/sample 2>/dev/null); do
```

Shellcheck (0.11.0)

[info] To read lines rather than words, pipe/redirect to a 'while read' loop.

(SC2013)